File: app/src/main/java/com/cielo/ordermanager/sdk/sample/ListOrdersActivity.java
package com.cielo.ordermanager.sdk.sample;

import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.cielo.ordermanager.sdk.R;
import com.cielo.ordermanager.sdk.RecyclerViewEmptySupport;
import com.cielo.ordermanager.sdk.adapter.OrderRecyclerViewAdapter;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

import cielo.orders.domain.Credentials;
import cielo.orders.domain.Order;
import cielo.orders.domain.ResultOrders;
import cielo.sdk.order.OrderManager;
import cielo.sdk.order.ServiceBindListener;

public class ListOrdersActivity extends AppCompatActivity {
    OrderManager orderManager;
    private final String TAG = "ORDER_LIST";
    private RecyclerViewEmptySupport recyclerView;
    private TextView txtEmptyValue;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_orders);
        recyclerView = findViewById(R.id.recycler_view);
        txtEmptyValue = findViewById(R.id.empty_view);
        recyclerView.addItemDecoration(new DividerItemDecoration(recyclerView.getContext(),
                DividerItemDecoration.VERTICAL));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setTitle("Listagem de Ordens");
        configSDK();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_refresh, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.retrieveOrder:
                retrieveOrder();
                return true;
            case R.id.refreshOrders:
                refreshOrdersFromServer();
                return true;
            default:
                finish();
                return super.onOptionsItemSelected(item);
        }
    }

    public void configSDK() {
        Credentials credentials = new Credentials("rSAqNPGvFPJI", "XZevoUYKmkVr");
        orderManager = new OrderManager(credentials, this);
        orderManager.bind(this, new ServiceBindListener() {
            @Override
            public void onServiceBoundError(Throwable throwable) {
                Toast.makeText(getApplicationContext(), "Erro fazendo bind do serviço de ordem -> " + throwable.getMessage(), Toast.LENGTH_LONG).show();
            }

            @Override
            public void onServiceBound() {
                retrieveOrder();
            }

            @Override
            public void onServiceUnbound() {
                // Implementação do serviço desvinculado, se necessário
            }
        });
    }

    private void retrieveOrder() {
        try {
            txtEmptyValue.setText(R.string.empty_orders);
            recyclerView.setEmptyView(txtEmptyValue);
            recyclerView.setLayoutManager(new LinearLayoutManager(this));
            final List<Order> orderList = retrieveOrdersFromRepository();
            recyclerView.setAdapter(new OrderRecyclerViewAdapter(orderList));
            for (Order or : orderList) {
                Log.i("Order: ", or.getNumber() + " - " + or.getPrice());
            }
        } catch (UnsupportedOperationException e) {
            Toast.makeText(this, "FUNCAO NAO SUPORTADA NESSA VERSAO DA LIO", Toast.LENGTH_LONG).show();
        }
    }

    private List<Order> retrieveOrdersFromRepository() {
        List<Order> allOrders = new ArrayList<>();
        ResultOrders resultOrders;
        for (int currentPage = 0; (resultOrders = orderManager.retrieveOrders(50, currentPage)) != null
                && currentPage < resultOrders.getTotalPages(); currentPage++) {
            allOrders.addAll(resultOrders.getResults());
        }
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            allOrders.sort(Comparator.comparing(Order::getReleaseDate, Comparator.nullsLast(Comparator.naturalOrder()))
                    .thenComparing(Order::getCreatedAt, Comparator.nullsLast(Comparator.naturalOrder())).reversed());
        }
        return allOrders;
    }

    private void refreshOrdersFromServer() {
        orderManager.refreshOrdersFromServer();
    }
}

File: app/src/main/java/com/cielo/ordermanager/sdk/sample/ListOrdersActivity.kt
package com.cielo.ordermanager.sdk.sample

import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.widget.TextView
import android.widget.Toast
import cielo.orders.domain.Order
import com.cielo.ordermanager.sdk.R
import com.cielo.ordermanager.sdk.TAG
import com.cielo.ordermanager.sdk.adapter.OrderRecyclerViewAdapter
import kotlinx.android.synthetic.main.activity_orders.recyclerView

class ListOrdersActivity : SdkPaymentActivity() {
    var txtEmptyValue: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_orders)
        supportActionBar?.run {
            setDisplayHomeAsUpEnabled(true)
            title = "LisTAGem de Ordens"
        }
        configSDK(this::listOrders)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.menu_refresh, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.refresh -> {
                listOrders()
                true
            }
            else -> {
                finish()
                super.onOptionsItemSelected(item)
            }
        }
    }

//    private fun configSDK() {
//        val credentials = Credentials(getString(R.string.cielo_client_id),
//                getString(R.string.cielo_access_token))
//        orderManager = OrderManager(credentials, this)
//        orderManager?.bind(this, object : ServiceBindListener {
//            override fun onServiceBoundError(throwable: Throwable) {
//                Toast.makeText(applicationContext, String.format("Erro fazendo bind do serviço de ordem -> %s",
//                        throwable.message), Toast.LENGTH_LONG).show()
//            }
//
//            override fun onServiceBound() {
//                listOrders()
//            }
//
//            override fun onServiceUnbound() {}
//        })
//    }

    private fun listOrders() {
        try {
            val resultOrders = orderManager?.retrieveOrders(200, 0)
            txtEmptyValue?.setText(R.string.empty_orders)
            //            recyclerView.setEmptyView(txtEmptyValue);
            if (resultOrders != null) {
                recyclerView?.layoutManager = androidx.recyclerview.widget.LinearLayoutManager(this@ListOrdersActivity)
                val orderList: List<Order> = resultOrders.results
                recyclerView?.adapter = OrderRecyclerViewAdapter(orderList)
                Log.i(TAG, "orders: $orderList")
                for (or in orderList) {
                    Log.i("Order: ", or.number + " - " + or.price)
                }
            }
        } catch (e: UnsupportedOperationException) {
            Toast.makeText(this@ListOrdersActivity, "FUNCAO NAO SUPORTADA NESSA VERSAO DA LIO",
                    Toast.LENGTH_LONG).show()
        }
    }
}
